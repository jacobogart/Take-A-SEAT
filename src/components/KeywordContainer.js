import React, { Component } from 'react';
import Keyword from './Keyword'

class KeywordContainer extends Component {

  render() {
    return (
      <section className="KeywordContainer">
        <p>Click once for details, click again to select.</p>
        <div className="button-holder">
          {this.props.nextWords.map(nextWord => (
            <Keyword
              key={nextWord.id}
              keywordData={nextWord}
              findNextWords={this.props.findNextWords}
              chalkboardChecker={this.props.chalkboardChecker}
            />
          ))}
        </div> 
      </section>
    );
  }
}

export default KeywordContainer;
