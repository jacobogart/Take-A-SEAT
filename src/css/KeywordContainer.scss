.KeywordContainer {
  grid-area: keywords;
  margin-left: 10px;
  height: 100%;
  p {
    font-family: $script-font;
    color: white;
    margin: 20px;
    font-size: 24px;
  }
}

.button-holder{
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-content: center;
  height: 55vh;
}