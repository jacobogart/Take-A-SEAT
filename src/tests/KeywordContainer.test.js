import React from "react";
import KeywordContainer from "../components/KeywordContainer";
import { shallow } from "enzyme";

const mockChalkboardChecker = jest.fn();
const mockFindNextWords = jest.fn();
const mockKeywords = [];
const mockNextWords = [{
  "word": "const",
  "nextWords": [
    "mockData Name",
    "mockPropFunction"
  ],
  "value": "const ",
  "isEditable": false,
  "phase": ["S"],
  "details": "Declare a variable that cannot be reassigned",
  "id": 100
}]

describe('KeywordContainer', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <KeywordContainer
        keywords={mockKeywords}
        chalkboardChecker={mockChalkboardChecker}
        nextWords={mockNextWords}
        findNextWords={mockFindNextWords}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
