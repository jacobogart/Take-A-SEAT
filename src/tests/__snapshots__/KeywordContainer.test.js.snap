// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordContainer should match snapshot 1`] = `
<section
  className="KeywordContainer"
>
  <Keyword
    chalkboardChecker={[MockFunction]}
    key="100"
    keywordData={
      Object {
        "details": "Declare a variable that cannot be reassigned",
        "id": 100,
        "isEditable": false,
        "nextWords": Array [
          "mockData Name",
          "mockPropFunction",
        ],
        "phase": Array [
          "S",
        ],
        "value": "const ",
        "word": "const",
      }
    }
    updateCurrentWord={[Function]}
  />
</section>
`;
